changelog/8118.improvement.rst

```diff
@@ -0,0 +1 @@
+Added printing of ``os.PathLike`` objects in verbose output when used as an argument to ``@pytest.mark.parametrize``
```

Comment:
The nodeid has an effect on more than the verbose output. I suggest something more like this, you can tweak it of course:

```suggestion
When an object implementing :class:`os.PathLike` is used in parametrization, and an explicit parameter id isn't specified, its :func:`os.fspath` is now used for the id instead of an auto-generated ``{param name}{param index}`` id.
```